Sources/Workspace/Workspace.swift

```diff
@@ -1317,6 +1317,40 @@ extension Workspace {
         }
     }
 
+    /// Loads a single package in the context of a previously loaded graph. This can be useful for incremental loading in a longer-lived program, like an IDE.
+    public func loadPackage(
+        with identity: PackageIdentity,
+        packageGraph: PackageGraph,
+        observabilityScope: ObservabilityScope,
+        completion: @escaping(Result<Package, Error>) -> Void) {
+            guard let previousPackage = packageGraph.packages.first(where: { $0.identity == identity }) else {
+                return completion(.failure(StringError("could not find package with identity \(identity)")))
+            }
+
+            self.loadManifest(packageIdentity: identity,
+                              packageKind: previousPackage.underlyingPackage.manifest.packageKind,
+                              packagePath: previousPackage.path,
+                              packageLocation: previousPackage.underlyingPackage.manifest.packageLocation,
+                              observabilityScope: observabilityScope) { result in
+                let result = result.tryMap { manifest -> Package in
+                    let builder = PackageBuilder(
+                        identity: identity,
+                        manifest: manifest,
+                        productFilter: .everything,
```

Comment:
This parameter will currently not match the previous graph, but I think that's currently OK since we are not enabling the `ENABLE_TARGET_BASED_DEPENDENCY_RESOLUTION` flag.

Reply:
shell we add a code comment / TODO stating that here?